vnext/Microsoft.ReactNative.IntegrationTests/ReactNativeHostTests.cpp

```diff
@@ -41,48 +47,37 @@ TEST_CLASS (ReactNativeHostTests) {
-  //   (__cdecl*)(winrt::Windows::ApplicationModel::Core::ICoreImmersiveApplication const &)> 12
-  //   Microsoft_ReactNative!winrt::impl::call_factory_cast<winrt::Windows::ApplicationModel::Core::CoreApplicationView
-  //   (__cdecl*)(winrt::Windows::ApplicationModel::Core::ICoreImmersiveApplication const
-  //   &),winrt::Windows::ApplicationModel::Core::CoreApplication,winrt::Windows::ApplicationModel::Core::ICoreImmersiveApplication,<lambda_784b37f6ced458740b7689c626bf1066>
-  //   > 13 Microsoft_ReactNative!winrt::Windows::ApplicationModel::Core::CoreApplication::MainView 14
-  //   Microsoft_ReactNative!Mso::DispatchQueueStatic::MakeMainUIScheduler 15
-  //   Microsoft_ReactNative!Mso::DispatchQueueStatic::MainUIQueue 16
-  //   Microsoft_ReactNative!Mso::DispatchQueue::MainUIQueue 17
-  //   Microsoft_ReactNative!Mso::React::ReactInstanceWin::InitUIMessageThread 18
-  //   Microsoft_ReactNative!Mso::React::ReactInstanceWin::Initialize
-  // In the short term, there are several ways in which this could get mitigated (e.g. transforming the integration test
-  // app into a UWP app). In the long term, the issue should get addressed by splitting the Microsoft.ReactNative.dll
-  // into a "core" and a "UI" DLL, and by having the tests in this project target the "core" DLL.
-  SKIPTESTMETHOD(JsFunctionCall_Succeeds) {
+  TEST_METHOD(JsFunctionCall_Succeeds) {
     std::future<TestHostModule &> testHostModule = TestHostModule::Instance.get_future();
     std::future<int> returnValue = TestHostModule::IntReturnValue.get_future();
 
     winrt::Microsoft::ReactNative::ReactNativeHost host{};
-    host.PackageProviders().Append(winrt::make<TestPackageProvider>());
 
-    // bundle is assumed to be co-located with the test binary
-    wchar_t testBinaryPath[MAX_PATH];
-    TestCheck(GetModuleFileNameW(NULL, testBinaryPath, MAX_PATH) < MAX_PATH);
-    testBinaryPath[std::wstring_view{testBinaryPath}.rfind(L"\\")] = 0;
-    host.InstanceSettings().BundleRootPath(testBinaryPath);
+    auto queueController = winrt::Windows::System::DispatcherQueueController::CreateOnDedicatedThread();
+    queueController.DispatcherQueue().TryEnqueue([&]() noexcept {
+      host.PackageProviders().Append(winrt::make<TestPackageProvider>());
+
+      // bundle is assumed to be co-located with the test binary
+      wchar_t testBinaryPath[MAX_PATH];
+      TestCheck(GetModuleFileNameW(NULL, testBinaryPath, MAX_PATH) < MAX_PATH);
+      testBinaryPath[std::wstring_view{testBinaryPath}.rfind(L"\\")] = 0;
 
-    host.InstanceSettings().JavaScriptBundleFile(L"AddValues.bundle");
-    host.ReloadInstance();
+      host.InstanceSettings().BundleRootPath(testBinaryPath);
+      host.InstanceSettings().JavaScriptBundleFile(L"AddValues");
+      host.InstanceSettings().UseDeveloperSupport(false);
+      host.InstanceSettings().UseWebDebugger(false);
+      host.InstanceSettings().UseFastRefresh(false);
+      host.InstanceSettings().UseLiveReload(false);
```

Comment:
I've noticed that also in the template these are all set at the same time, should we have a single method to toggle these settings (in case we get more in the future) i.e.
`UseDevSettings(true|false)`

Reply:
It is a good point. We could implement it in a separate PR.

---
In reply to: [432968392](https://github.com/microsoft/react-native-windows/pull/5071#discussion_r432968392) [](ancestors = 432968392)

Reply:
I have created Issue #5078 to address it in the future.

---
In reply to: [433245285](https://github.com/microsoft/react-native-windows/pull/5071#discussion_r433245285) [](ancestors = 433245285,432968392)